dev/ToastNotifications/ToastNotificationUtility.cpp

```diff
@@ -0,0 +1,190 @@
+﻿
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "ToastNotificationUtility.h"
+#include <winrt/Windows.ApplicationModel.Core.h>
+#include <winrt/Windows.Foundation.h>
+#include <externs.h>
+
+namespace winrt
+{
+    using namespace winrt::Windows::Foundation;
+    using namespace Windows::ApplicationModel::Core;
+}
+
+std::wstring RetrieveUnpackagedToastGuid()
+{
+    wil::unique_cotaskmem_string processName;
+    THROW_IF_FAILED(GetCurrentProcessPath(processName));
+
+    std::wstring wideStringProcessName{ processName.get() };
+    std::wstring subKey{ c_appIdentifierPath + ConvertPathToKey(wideStringProcessName) };
+
+    wil::unique_hkey hKey;
+    THROW_IF_FAILED(RegCreateKeyEx(
+        HKEY_CURRENT_USER,
+        subKey.c_str(),
+        0,
+        nullptr,
+        REG_OPTION_NON_VOLATILE,
+        KEY_ALL_ACCESS,
+        nullptr,
+        &hKey,
+        nullptr));
+
+    WCHAR registeredGuidBuffer[GUID_LENGTH + 3]; // GUID length + '{' + '}' + '\0'
+    DWORD bufferLength = sizeof(registeredGuidBuffer);
+    HRESULT status = RegGetValueW(
+        hKey.get(),
+        nullptr,
+        L"ToastGUID",
+        RRF_RT_REG_SZ,
+        nullptr,
+        &registeredGuidBuffer,
+        &bufferLength);
+
+    if (status == ERROR_FILE_NOT_FOUND)
+    {
+        GUID newToastGuid;
+        THROW_IF_FAILED(CoCreateGuid(&newToastGuid));
+
+        wil::unique_cotaskmem_string newToastGuidString;
+        THROW_IF_FAILED(StringFromCLSID(newToastGuid, &newToastGuidString));
+
+        std::wstring guidWideStr{ newToastGuidString.get() };
+        RegisterValue(hKey, L"ToastGUID", reinterpret_cast<const BYTE*>(guidWideStr.c_str()), REG_SZ, guidWideStr.size() * sizeof(wchar_t));
+        return guidWideStr;
+    }
+    else
+    {
+        THROW_HR_IF(status, FAILED_WIN32(status));
+    }
+    return registeredGuidBuffer;
+}
+
+std::wstring RetrieveToastGuid()
```

Comment:
The method name is incorrect. It does not necessarily return a GUID. It should be called RetrieveAppId()